dotCMS/src/main/java/com/dotcms/mock/request/MockHttpRequest.java

```diff
@@ -18,15 +26,36 @@
 public class MockHttpRequest implements MockRequest {
 
     private final HttpServletRequest request;
-
+    private final Map<String,String[]> paramMap;
     public MockHttpRequest(final String hostname, final String uri) {
         HttpServletRequest mockReq = new BaseRequest().request();
         Mockito.when(mockReq.getRequestURI()).thenReturn(uri);
         Mockito.when(mockReq.getRequestURL()).thenReturn(new StringBuffer("http://" + hostname + uri));
         Mockito.when(mockReq.getServerName()).thenReturn(hostname);
         Mockito.when(mockReq.getRemoteAddr()).thenReturn("127.0.0.1");
         Mockito.when(mockReq.getRemoteHost()).thenReturn("127.0.0.1");
+        paramMap = new HashMap<>();
+        if(uri.contains("?")) {
+            final String queryString = uri.substring(uri.indexOf("?") + 1, uri.length());
+            Mockito.when(mockReq.getQueryString()).thenReturn(queryString);
+            List<NameValuePair> additional = URLEncodedUtils.parse(queryString, Charset.forName("UTF-8"));
+            for(NameValuePair nvp : additional) {
+                paramMap.compute(nvp.getName(), (k, v) -> (v == null) ? new String[] {nvp.getValue()} : new String[]{nvp.getValue(),v[0]});
+            }
+            Mockito.when(mockReq.getQueryString()).thenReturn(queryString);
+            Mockito.when(mockReq.getParameterMap()).thenReturn(paramMap);
+            Mockito.when(mockReq.getParameterNames()).thenReturn(Collections.enumeration(paramMap.keySet()));
+            Mockito.when(mockReq.getParameter(Mockito.anyString())).thenAnswer(new Answer<String>() {
+                public String answer(InvocationOnMock invocation) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'invocation' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5391667)